File: LeetCode_2016/Main.java
package LeetCode_2016;

class Solution {

    public int maximumDifference(int[] nums) {
        int n = nums.length;
        int ans = -1, premin = nums[0];
        for (int i = 1; i < n; ++i) {
            if (nums[i] > premin) {
                ans = Math.max(ans, nums[i] - premin);
            } else {
                premin = nums[i];
            }
        }
        return ans;
    }
}

public class Main {
    
}

File: LeetCode_2016/Main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumDifference = function (nums) {
    const n = nums.length;
    let ans = -1, premin = nums[0];
    
    for (let i = 1; i < n; ++i) {
        if (nums[i] > premin) {
            ans = (ans < nums[i] - premin ? nums[i] - premin : ans);
        } else {
            premin = nums[i];
        }
    }
    return ans;
};

console.log(maximumDifference([7, 1, 5, 4]));